src/components/redux/modules/booksSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState: [],
+  reducers: {
+    bookAdded(state, action) {
+      state.push({
+        id: action.payload.id,
+        text: action.payload.title,
+        author: action.payload.author,
+      });
+    },
+    progressUpdate(state, action) {
+      const book = state.find((book) => book.id === action.payload);
+      book.progress += 1;
+    },
+  },
+});
+
+export const { bookAdded, progressUpdate } = booksSlice.actions;
+export default booksSlice.reducer;
```

Comment:
Great job creating a Redux Slice 😄! As requested in the project requirements: 
![image](https://user-images.githubusercontent.com/42799579/208785456-c18ae085-6d24-4c33-9644-7c2833e04bc0.png) 

Kindly consider that you still need to implement some of Redux's functionalities 😉. Please make sure to: 
- [ ] Use [Redux Thunk](https://github.com/reduxjs/redux-thunk) to connect with your API. Currently, your project isn't adding or removing books. Keep in mind that you need to keep the functionality after refactoring your code with Redux's tools. 

- [ ] Refactor your _add book_ and _remove book_ to persist changes in the server. This means that you should hit the API to update your book list after adding & removing any book. So, when the user comes back later, the page will display the updated book list 😉. 

> Check more about the Bookstore API [on this link](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739).

Reply:
Thanks @Gopxfs  updated accordingly